scripts/provisioning/upgradecmd.py

```diff
@@ -0,0 +1,186 @@
+                    self.get_confkey('S3_KEYSTORE_CONFIG_SAMPLE_FILE'),
+                    self.get_confkey('S3_KEYSTORE_CONFIG_UNSAFE_ATTR_FILE'),
+                    self.get_confkey('S3_BGDELETE_CONFIG_SAMPLE_FILE'),
+                    self.get_confkey('S3_BGDELETE_CONFIG_UNSAFE_ATTR_FILE'),
+                    self.get_confkey('S3_CLUSTER_CONFIG_SAMPLE_FILE'),
+                    self.get_confkey('S3_CLUSTER_CONFIG_UNSAFE_ATTR_FILE')]
+
+    # copy all the config files from the /opt/seagate/cortx to /etc/cortx
+    for config_file in config_files:
+      self.logger.info(f"Source config file: {config_file}")
+      dest_config_file = config_file.replace("/opt/seagate/cortx", "/etc/cortx")
+      self.logger.info(f"Dest config file: {dest_config_file}")
+      os.makedirs(os.path.dirname(dest_config_file), exist_ok=True)
+      shutil.move(config_file, dest_config_file)
+      self.logger.info("Config file copied successfully to /etc/cortx")
+
+  def service_based_check_and_copy(self, service_list):
+    """ Copy sample and unsafe attribute config files from /opt/seagate/cortx to /etc/cortx."""
+    lookup_service = { "io"           : ["cluster", "s3"],
+                       "auth"         : ["cluster", "keystore", "auth"],
+                       "bg_consumer"  : ["cluster", "bgdelete"],
+                       "bg_producer"  : ["cluster", "bgdelete"]
+    }
+    old_files = {'s3'       : os.path.join(self.s3_tmp_dir, "s3config.yaml.sample.old"),
+                 'auth'     : os.path.join(self.s3_tmp_dir, "authserver.properties.sample.old"),
+                 'keystore' : os.path.join(self.s3_tmp_dir, "keystore.properties.sample.old"),
+                 'bgdelete' : os.path.join(self.s3_tmp_dir, "config.yaml.sample.old"),
+                 'cluster'  : os.path.join(self.s3_tmp_dir, "s3_cluster.yaml.sample.old")
+    }
+    config_files = {'s3'       : self.get_confkey('S3_CONFIG_SAMPLE_FILE'),
+                    'auth'     : self.get_confkey('S3_AUTHSERVER_CONFIG_SAMPLE_FILE'),
+                    'keystore' : self.get_confkey('S3_KEYSTORE_CONFIG_SAMPLE_FILE'),
+                    'bgdelete' : self.get_confkey('S3_BGDELETE_CONFIG_SAMPLE_FILE'),
+                    'cluster'  : self.get_confkey('S3_CLUSTER_CONFIG_SAMPLE_FILE')
+    }
+
+    for ser_key in service_list:
+      ser_list = lookup_service[ser_key]
+      for item in ser_list:
+        # check necessary files before calling merge
+        sample_old_file = old_files[item]
+        if not os.path.exists(sample_old_file):
+          self.logger.info(f"{sample_old_file} backup file not found")
+          raise S3PROVError(f'process: {self.name} failed')
+        # overwrite /opt/seagate/cortx/*/*.sample
+        # to /etc/cortx/*/*.sample
+        s3_sample_file = config_files[item]
+        s3_file = s3_sample_file.replace("/opt/seagate/cortx", self.base_config_file_path)
+        shutil.copy(s3_sample_file, s3_file)
+
+  def post_merge_copy(self):
```

Comment:
Move this to setupcmd so that it is available as common function, call it from both places..
Rename it to make_sample_old_files

Reply:
Done.